web-api/src/storage/postgres.rs

```diff
@@ -622,19 +656,187 @@ impl Storage {
     }
 }
 
+#[cfg(feature = "ET-3837")]
 #[derive(FromRow)]
-struct QueriedCoi {
-    coi_id: CoiId,
-    is_positive: bool,
-    embedding: NormalizedEmbedding,
-    /// The count is a `usize` stored as `i32` in database
-    view_count: i32,
-    /// The time is a `u64` stored as `i64` in database
-    view_time_ms: i64,
-    last_view: DateTime<Utc>,
+struct QueriedDocumentProperties(Json<DocumentProperties>);
+
+#[cfg(feature = "ET-3837")]
+#[async_trait(?Send)]
+impl storage::DocumentProperties for Storage {
+    async fn get(&self, id: &DocumentId) -> Result<Option<DocumentProperties>, Error> {
+        let mut tx = self.postgres.pool.begin().await?;
+
+        let properties = sqlx::query_as::<_, QueriedDocumentProperties>(
+            "SELECT properties
+            FROM document
+            WHERE document_id = $1;",
+        )
+        .bind(id)
+        .fetch_optional(&mut tx)
```

Comment:
:wrench:  You can just pass the pool instead of an transaction the always using of a transaction was only a thing due to a subtle bug of the sqlite driver of sqlx.

Reply:
Same for some of the other endpoints below.